src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -266,6 +266,36 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetTags(NormalizedSnapshotSpanC
                 }
             }
 
+            /// <summary>
+            /// Force computation of up to date tags.
+            /// This is called by the editor for scenarios such as print or rich text copy with classifications enabled.
+            /// Note that the editor will block the UI until results are returned.
+            /// </summary>
+            public IEnumerable<ITagSpan<IClassificationTag>> GetAllTags(NormalizedSnapshotSpanCollection spans, CancellationToken cancellationToken)
+            {
+                if (spans.IsEmpty() || _workspace == null)
+                {
+                    return SpecializedCollections.EmptyEnumerable<ITagSpan<IClassificationTag>>();
+                }
+
+                var document = _subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
```

Comment:
This should be using the snapshot that comes from spans; there's no guarantee the snapshot being asked about matches the snapshot the buffer is currently on.